languages/go/reference/exercise-concepts/parallel-letter-frequency.md

```diff
@@ -0,0 +1,29 @@
+# Concepts of Parallel Letter Frequency
+
+## Concurrency
+
+The goal of this exercise is to learn to write concurrent code to count the letters inside different texts.
+
+## Goroutines
+
+In order to count concurrent the apppearance of different letters in separate texts it is important to be able to create goroutines with the `go` keyword, one for each text where the letters are counted.
+
```

Comment:
This is also the first core exercise (I think) that requires tests to be run with the race detector. This trips up a lot of students. When they get a data race, it's a good opportunity to explain "Don't communicate by sharing data, share data by communicating". I sometimes explain the 'guard' pattern (access to the contended data is reserved for a single goroutine, often the main goroutine, which mediates requests from workers to read or modify the data).

Reply:
Seems to me you'd like to see some concept exercises teach patterns in Go as well. I've also thought about that already. I will create a section for that in out concept list.

Is 'guard' pattern a general term for this? In this link it is mentioned as "protecting data by `confinement`": https://www.oreilly.com/library/view/concurrency-in-go/9781491941294/ch04.html